sensor_msgs/include/sensor_msgs/impl/point_cloud2_iterator.hpp

```diff
@@ -125,12 +125,19 @@ inline void PointCloud2Modifier::reserve(size_t size)
 
 inline void PointCloud2Modifier::resize(size_t size)
 {
-  cloud_msg_.data.resize(size * cloud_msg_.point_step);
+  size_t original_size = cloud_msg_.height * cloud_msg_.width;
+  if (original_size == size) {
```

Comment:
I think the return should happen after the `cloud_msg_.data.resize(total_size);` call but before the `// Update height/width`